clang/test/Driver/sycl-offload-intelfpga.cpp

```diff
@@ -31,7 +31,9 @@
 // CHK-FPGA-LINK: clang-offload-bundler{{.*}} "-type=o" "-targets=sycl-spir64_fpga-unknown-unknown-sycldevice" "-inputs=[[INPUT:.+\.o]]" "-outputs=[[OUTPUT1:.+\.o]]" "-unbundle"
 // CHK-FPGA-LINK-NOT: clang-offload-bundler{{.*}}
 // CHK-FPGA-LINK: llvm-link{{.*}} "[[OUTPUT1]]" "-o" "[[OUTPUT2_1:.+\.bc]]"
-// CHK-FPGA-LINK: sycl-post-link{{.*}} "-ir-output-only" "-spec-const=default" "-o" "[[OUTPUT2:.+\.bc]]" "[[OUTPUT2_1]]"
+// CHK-FPGA-LINK: sycl-post-link
+// CHK-FPGA-LINK-NOT: -split-esimd
```

Comment:
If I'm not misinterpreting this, the following approach would be slightly more preferable in this place & similar ones:
```suggestion
// CHK-FPGA-LINK: sycl-post-link{{.*}} "-split-esimd"
```

Reply:
I'm not sure this will work. I'm trying to check that there is no `-split-esimd` option in the invocation of `sycl-post-link`.

Reply:
True, however the same (more concise) approach is used in another test below at https://github.com/intel/llvm/pull/3086/files#diff-51b6d200f5191530d536b4795db699c5481490d74eb85505ddc8f7389b76cba4R318.

Reply:
Yes, but here I also need to check that other options like `-ir-output-only` are passed to the `sycl-post-link`.

Reply:
I guess something like this works at [line 30](https://github.com/intel/llvm/pull/3086/files#diff-b68febe70a8ad31e6528d1207bdf250ccc3e1b25fe08d3b871ac76cba13f4e6dR30) of this file. Anyhow, this is getting too nitty on my side, so let's just wrap this with whatever approach you find more self-explanatory.

Reply:
Actually, the check you mentioned is wrong. It doesn't test what it is intended to test. It doesn't test that there is no option `-check-section` passed to `clang-offload-bundler`. It tests that between the beginning of the file and the first `clang-offload-bundler{{.*}} -type=o` command there is no separate `clang-offload-bundler` command with `-check-section` option. I checked it locally.

Making it this way:
```
// CHK-FPGA-LINK-NOT: sycl-post-link{{.*}} "-split-esimd"
// CHK-FPGA-LINK: "-ir-output-only" ...
```
allows inputs when 1) there is no `sycl-post-link` command and 2) `sycl-post-link` and the rest of the command options are on different lines.
So, this was the best option I was able to come up with.